ports/zephyr/modzephyr.c

```diff
@@ -79,6 +92,238 @@ STATIC MP_DEFINE_CONST_FUN_OBJ_0(mod_shell_net_iface_obj, mod_shell_net_iface);
+        mp_raise_ValueError("disk not found");
+    }
+
+    if (disk_access_ioctl(self->pdrv, DISK_IOCTL_GET_SECTOR_SIZE, &self->block_size)) {
+        mp_raise_ValueError("unable to get sector size");
+    }
+
+    return MP_OBJ_FROM_PTR(self);
+}
+
+STATIC mp_obj_t zephyr_disk_readblocks(mp_obj_t self_in, mp_obj_t block_num, mp_obj_t buf) {
+    zephyr_disk_obj_t *self = self_in;
+    mp_buffer_info_t bufinfo;
+    int ret;
+
+    mp_get_buffer_raise(buf, &bufinfo, MP_BUFFER_WRITE);
+    ret = disk_access_read(self->pdrv, bufinfo.buf, mp_obj_get_int(block_num), bufinfo.len / self->block_size);
+    return mp_obj_new_bool(ret == 0);
+}
+
+STATIC MP_DEFINE_CONST_FUN_OBJ_3(zephyr_disk_readblocks_obj, zephyr_disk_readblocks);
+
+STATIC mp_obj_t zephyr_disk_writeblocks(mp_obj_t self_in, mp_obj_t block_num, mp_obj_t buf) {
+    zephyr_disk_obj_t *self = self_in;
+    mp_buffer_info_t bufinfo;
+    int ret;
+
+    mp_get_buffer_raise(buf, &bufinfo, MP_BUFFER_READ);
+    ret = disk_access_write(self->pdrv, bufinfo.buf, mp_obj_get_int(block_num), bufinfo.len / self->block_size);
+    return mp_obj_new_bool(ret == 0);
+}
+
+STATIC MP_DEFINE_CONST_FUN_OBJ_3(zephyr_disk_writeblocks_obj, zephyr_disk_writeblocks);
+
+STATIC mp_obj_t zephyr_disk_ioctl(mp_obj_t self_in, mp_obj_t cmd_in, mp_obj_t arg_in) {
+    zephyr_disk_obj_t *self = self_in;
+    mp_int_t cmd = mp_obj_get_int(cmd_in);
+    int buf;
+
+    switch (cmd) {
+        case MP_BLOCKDEV_IOCTL_INIT:
+        case MP_BLOCKDEV_IOCTL_DEINIT:
+            return MP_OBJ_NEW_SMALL_INT(0);
+
+        case MP_BLOCKDEV_IOCTL_SYNC:
+            if (disk_access_ioctl(self->pdrv, DISK_IOCTL_CTRL_SYNC, &buf)) {
+                mp_raise_ValueError("unable to sync");
+            }
+            return MP_OBJ_NEW_SMALL_INT(0);
+
+        case MP_BLOCKDEV_IOCTL_BLOCK_COUNT:
+            if (disk_access_ioctl(self->pdrv, DISK_IOCTL_GET_SECTOR_COUNT, &buf)) {
+                mp_raise_ValueError("unable to get block count");
+            }
+            return MP_OBJ_NEW_SMALL_INT(buf);
+
+        case MP_BLOCKDEV_IOCTL_BLOCK_SIZE:
+            if (disk_access_ioctl(self->pdrv, DISK_IOCTL_GET_SECTOR_SIZE, &buf)) {
+                mp_raise_ValueError("unable to get block size");
+            }
+            return MP_OBJ_NEW_SMALL_INT(buf);
+
+        default:
+            return MP_OBJ_NEW_SMALL_INT(-1);
+    }
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_3(zephyr_disk_ioctl_obj, zephyr_disk_ioctl);
+
+STATIC const mp_rom_map_elem_t zephyr_disk_locals_dict_table[] = {
+    { MP_ROM_QSTR(MP_QSTR_readblocks), MP_ROM_PTR(&zephyr_disk_readblocks_obj) },
+    { MP_ROM_QSTR(MP_QSTR_writeblocks), MP_ROM_PTR(&zephyr_disk_writeblocks_obj) },
+    { MP_ROM_QSTR(MP_QSTR_ioctl), MP_ROM_PTR(&zephyr_disk_ioctl_obj) },
+};
+
+STATIC MP_DEFINE_CONST_DICT(zephyr_disk_locals_dict, zephyr_disk_locals_dict_table);
+
+STATIC const mp_obj_type_t zephyr_disk_type = {
+    { &mp_type_type },
+    .name = MP_QSTR_Disk,
```

Comment:
Could this class/type be called `DiskAccess` instead to better map to the Zephyr name?  I have no strong feelings either way, but it seems that `Disk` is very generic and maybe `DiskAccess` signals to the user that it's a more special thing with a specific meaning.